test/Regex/characterclass_with_range.js

```diff
@@ -0,0 +1,132 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+WScript.LoadScriptFile("..\\UnitTestFramework\\UnitTestFramework.js");
+
+function testRegExp(str, regexp, expectedResult)
+{
+    actualResult = regexp.test(str);
+    errorMsg = "Expected result of test for match between string: '" + str + "' and regular expression: " + regexp + " to be " + 
+                    expectedResult + " but was " + actualResult;
+    assert.areEqual(expectedResult, actualResult, errorMsg); 
+}
+
+var tests = [
+    {
+        name : "RegExp tests with no flags",
+        body : function () 
+        {
+            let re = /[\s-a-z]/;
+            testRegExp("b", re, false);
+            testRegExp("a", re, true);
+            testRegExp(" ", re, true);
+            testRegExp("z", re, true);
+            testRegExp("\t", re, true);
+            testRegExp("q", re, false);
+            testRegExp("\\", re, false);
+            testRegExp("\u2028", re, true);
+            testRegExp("\u2009", re, true);
+        }
+    },
+    {
+        name : "RegExp tests with IgnoreCase flag set",
+        body : function () 
+        {
+            let reIgnoreCase = /^[\s-a-z]$/i;
+            testRegExp("O", reIgnoreCase, false);
+            testRegExp("A", reIgnoreCase, true);
+            testRegExp(" ", reIgnoreCase, true);
+            testRegExp("z", reIgnoreCase, true);
+            testRegExp("\t", reIgnoreCase, true);
+            testRegExp("\u2028", reIgnoreCase, true);
+            testRegExp("\u2009", reIgnoreCase, true);
+        }
+    },
+    {
+        name : "RegExp tests with Unicode flag set",
+        body : function () 
+        {
+            let reUnicode = /^[a-d]$/u;
+            testRegExp("a", reUnicode, true);
+            testRegExp("c", reUnicode, true);
+            testRegExp("d", reUnicode, true);
+            testRegExp("C", reUnicode, false);
+            testRegExp("g", reUnicode, false);
+            testRegExp("\u2028", reUnicode, false);
+            testRegExp("\u2009", reUnicode, false);
+            assert.throws(() => eval("/^[\\s-z]$/u.exec(\"-\")"), SyntaxError, "Expected an error due to character sets not being allowed in ranges when unicode flag is set.", "Character classes not allowed in class ranges");

```

Comment:
since this is in a function, is it necessary to eval the regex? (Also, if you need to construct a RegExp from a string, you could use the `RegExp` constructor instead of `eval`.)

Reply:
Using the literal regex or regex constructor will throw a SyntaxError when parsing the script.